readme-contents/AppInsightsClientTracking.md

```diff
@@ -0,0 +1,33 @@
+# App Insights Client Tracking
+
+## What is provided?
+
+By including the library you get basic App Insights client tracking added to your application.
+This includes:
+* a basic Spring `HandlerInterceptor`/`WebFilter` to perform the client tracking (depending upon the Spring app being servlet or reactive based)
+* `client_id` and `user_name` are extracted from JWTs and added to the App Insights `Properties` / `customDimensions` as `clientId` and `uername`
+
+> [!WARNING]
+> If you already have client tracking configured then this library tries to get out of your way and wil do nothing if you 
+> have beans/classes named `ClientTrackingConfiguration`, `ClientTrackingInterceptor` or `ClientTrackingWebFilter`.
+> This also means you cannot use these class names to configure this library because it will do nothing. 
```

Comment:
This is a little awkward. Many projects have already implemented their own client tracking config, and I wanted the library to get out of the way and do nothing in this case. However the only way I could think of to achieve this is to do nothing if we find an existing bean called `clientTrackingInterceptor` or `clientTrackingWebFilter`. Better ideas welcome!